spec/models/reports/region_summary_schema_spec.rb

```diff
@@ -76,37 +76,51 @@ def refresh_views
   describe "appointment scheduled days percentages" do
     it "returns percentages of appointments scheduled across months in a given range" do
       facility = create(:facility)
-      patient = create(:patient, assigned_facility: facility, recorded_at: 1.month.ago)
+      htn_patient = create(:patient, assigned_facility: facility, recorded_at: 1.month.ago)
+      diabetes_patient = create(:patient, :diabetes, assigned_facility: facility, recorded_at: 1.month.ago)
       range = Period.month(2.month.ago)..Period.current
-      _appointment_scheduled_0_to_14_days = create(:appointment, patient: patient, facility: facility, scheduled_date: 10.days.from_now, device_created_at: Date.today)
+      _htn_appointment_scheduled_0_to_14_days = create(:appointment, patient: htn_patient, facility: facility, scheduled_date: 10.days.from_now, device_created_at: Date.today)
+      _diabetes_appointment_scheduled_0_to_14_days = create(:appointment, patient: diabetes_patient, facility: facility, scheduled_date: 10.days.from_now, device_created_at: Date.today)
 
       refresh_views
 
       schema = described_class.new(Region.where(id: facility.region), periods: range)
       expect(schema.appts_scheduled_0_to_14_days_rates[facility.slug][range.first]).to eq(0)
       expect(schema.appts_scheduled_0_to_14_days_rates[facility.slug][range.to_a.second]).to eq(0)
       expect(schema.appts_scheduled_0_to_14_days_rates[facility.slug][range.last]).to eq(100)
+
+      expect(schema.diabetes_appts_scheduled_0_to_14_days_rates[facility.slug][range.first]).to eq(0)
+      expect(schema.diabetes_appts_scheduled_0_to_14_days_rates[facility.slug][range.to_a.second]).to eq(0)
+      expect(schema.diabetes_appts_scheduled_0_to_14_days_rates[facility.slug][range.last]).to eq(100)
     end
 
     it "returns percentages of appointments scheduled in a month in the given range" do
       facility = create(:facility)
-      patient = create(:patient, assigned_facility: facility, recorded_at: 4.month.ago)
```

Comment:
We're currently only testing the case where appointments are being scheduled across different months. Let's add a spec where we test multiple appointments created in the same month, but across buckets. This would test that we're passing in the right data to `rouded_percentages`